examples/npm_lib/src/main.rs

```diff
@@ -0,0 +1,62 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use log::debug;
+use wasm_bindgen::JsValue;
+use web_sys::{EventTarget, HtmlElement, HtmlInputElement as InputElement};
+
+use kobold::prelude::*;
+
+mod js;
+
+struct State {
+    hash: String,
+}
+
+impl State {
+    fn new() -> Self {
+        State {
+            hash: "0x0".to_owned(),
+        }
+    }
+}
+
+async fn onclick_pjs_process(state: Signal<State>, event: MouseEvent<HtmlElement>) {
+    let res: Result<wasm_bindgen::JsValue, wasm_bindgen::JsValue> =
+        js::browser_js::run_npm_lib().await;
+
+    state.update(move |state| match res {
+        Ok(jsv) => {
+            debug!("jsv {:?}", jsv);
+            match JsValue::as_string(&jsv) {
+                Some(hash) => {
+                    debug!("hash {:?}", hash);
+                    state.hash = hash;
+                }
+                None => panic!("error converting JsValue to String for hash"),
+            }
+        }
+        _ => panic!("error fetching from API"),
+    });
```

Comment:
Why not do error handling outside of the `update`?

```suggestion
    let res = js::browser_js::run_npm_lib().await;
        
    let hash = match res.ok().and_then(|value| value.as_string()) {
        Some(hash) => hash,
        None => panic!("error fetching from API"),
    };

    state.update(move |state| state.hash = hash);
```